### PR TITLE
Create QR page

### LINKED ISSUE #2187: [FEATURE] QR code page from profile
### Description

It would be great to have a page that generates a QR for people to use to share their profile link that others can scan

- link from profile page to unique QR code page
- QR code page to have only the name, bio, pic, and then the QR code big
- QR code to be url to users profile page

### Screenshots

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by furkanmutlu**
## Fixes Issue

Closes #2187 

## Changes proposed

- Used qrcode.react to generate Qr codes
- Created link from profile page to unique QR code page
- Created QR code page

## Check List (Check all the applicable boxes) 

- [ x] My code follows the code style of this project.
- [ ] My change requires changes to the documentation.
- [ ] I have updated the documentation accordingly.
- [ x] All new and existing tests passed.
- [ x] This PR does not contain plagiarized content.
- [ x] The title of my pull request is a short description of the requested changes.

## Screenshots

image
image

## Note to reviewers

<a href="https://gitpod.io/#https://github.com/EddieHubCommunity/LinkFree/pull/2233"></a>

### CODE REVIEW

File: pages/[username].js [hunk 1]
```diff
@@ -1,146 +1,155 @@
-import Head from "next/head";
-import ReactMarkdown from "react-markdown";
-import { abbreviateNumber } from "js-abbreviation-number";
+import Head from 'next/head';
```
Comment (eddiejaoude) [inline pages/[username].js:1]: I think the coding style of the project is `"`

File: pages/[username].js [hunk 2]
```diff
@@ -1,146 +1,155 @@
...
-    const res = await fetch(
-      `${app.baseUrl}/api/users/${context.query.username}`
-    );
-    data = await res.json();
-  } catch (e) {
-    console.log("ERROR user not found ", e);
-  }
+    let data = {};
+    try {
+        const res = await fetch(
```
Comment (eddiejaoude) [inline pages/[username].js:19]: I think the coding style of the project is 2 spaces not 4

It looks like there are more changes which makes it difficult to review

File: pages/qr/[username].js [hunk 3]
```diff
@@ -0,0 +1,104 @@
+import Head from 'next/head';
```
Comment (eddiejaoude) [inline pages/qr/[username].js:1]: would it be better to move the `pages/[username].js` file and this one to a single directory `pages/users/`?

Comment (furkanmutlu) [reply pages/qr/[username].js:1]: yes, I like this idea, I will push the changes

File: pages/[username].js (hunk 2 above)
Comment (furkanmutlu) [reply pages/[username].js:19]: I just changed some of my editor settings now, I will push changes soon

File: pages/[username].js (hunk 1 above)
Comment (furkanmutlu) [reply pages/[username].js]: Should be fixed now

File: pages/[username].js (hunk 2 above)
Comment (furkanmutlu) [reply pages/[username].js]: should be fixed now

File: pages/qr/[username].js (hunk 3 above)
Comment (furkanmutlu) [reply pages/qr/[username].js]: fixed

**Comment by furkanmutlu** [comment]
@eddiejaoude  I was thinking to reorganise this two pages.(`username` and `qr/username`) Since there is many repeating code,  I was thinking to create a `util` folder that for example fetches user data. Also it might be better to create a component for repeating part of their templates.(like Head, Portfolio)
Maybe we could do it under an another PR ? I would be happy to do.

I was also wondering how do you think should we resolve PR comment  after pushing fixed changes ?

**Comment by eddiejaoude** [comment]
Thanks for making the changes, going to test it out now locally. 

Yep more changes/refactors can come in another PR - I like PRs to be as small as possible so they are easier to review etc

> I was also wondering how do you think should we resolve PR comment after pushing fixed changes ?

What do you mean?

**Comment by eddiejaoude** [comment]
The profile url has now changed to `http://localhost:3000/users/eddiejaoude`, this cannot change, so maybe the folder idea is not such a good idea - let me check the docs again

**Comment by furkanmutlu** [comment]
> What do you mean?
I mean this:
image

Should I resolve it after changes, or do you prefer to do it by yourself

**Comment by eddiejaoude** [comment]
Sorry I think we need to move the file back so that it is on the original route

**Comment by furkanmutlu** [comment]
> 



> Sorry I think we need to move the file back so that it is on the original route

No worries, should be quick to update. 
So it will be as it was before right?

`http://localhost:3000/eddiejaoude`
`http://localhost:3000/qr/eddiejaoude`

**Comment by eddiejaoude** [comment]
> .Should I resolve it after changes, or do you prefer to do it by yourself

I still don't understand, could you elaborate please?

**Comment by eddiejaoude** [comment]
> No worries, should be quick to update.
> So it will be as it was before right?

> http://localhost:3000/eddiejaoude
> http://localhost:3000/qr/eddiejaoude

yes I think so 👍

**Comment by furkanmutlu** [comment]
@eddiejaoude updated the routes

**Comment by eddiejaoude** [comment]
I will merge to the `qr` branch and resolve the conflicts, as well as refactor to a single page - I think the QR can show/hide on the same page

**Review by eddiejaoude — APPROVED** [review]
Thank you for the contribution 👍 

I will merge to the qr branch and resolve the conflicts, as well as refactor to a single page - I think the QR can show/hide on the same page

**Comment by eddiejaoude** [comment]
PR continuation https://github.com/EddieHubCommunity/LinkFree/pull/2235

**Pull request merged**